src/main/java/jp/skypencil/errorprone/slf4j/CodepointIterator.java

```diff
@@ -0,0 +1,35 @@
+package jp.skypencil.errorprone.slf4j;
+
+import java.util.Iterator;
+
+final class CodepointIterator implements Iterator<Integer> {
+  private CharSequence sequence;
+  private int index;
+
+  CodepointIterator(CharSequence sequence) {
+    this.sequence = sequence;
+  }
+
+  @Override
+  public boolean hasNext() {
+    return index < sequence.length();
+  }
+
+  @Override
+  public Integer next() {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Add a "NoSuchElementException" for iteration beyond the end of the collection. ([squid:S2272](https://sonarcloud.io/organizations/kengotoda-github/rules#rule_key=squid%3AS2272))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=jp.skypencil.errorprone.slf4j%3Aerrorprone-slf4j&pullRequest=5&issues=AWRnohN2Mt9UOhrw_WLZ&open=AWRnohN2Mt9UOhrw_WLZ)

[ISSUE_KEY]: AWRnohN2Mt9UOhrw_WLZ